### DISCUSSION TITLE
Introduction: Computational biology student attempting open-source development!

### DISCUSSION THREAD

**Discussion by SampurnaM** [MDAnalysis/mdanalysis#4484] [GSoC Discussions]
- Hi MDAnalysis team,
I am currently writing my MPhil thesis on antigen-antibody interactions in organ transplantation using computational modelling tools. I am familiar with GROMACS simulations and **protein-protein interactions** have been the focus for me for the last few years. I am a trained biologist turned bioinformatician.
## My journey with MDAnalysis so far:
During my fellowship at EMBL-EBI, I was introduced to Jupyter Notebooks ( I am obsessed) and wanted to make an end-to-end Python pipeline for setup ,production and analysis of simulations.
The analysis bit is where I came across the beautiful, beautiful thing that is MDAnalysis, and for non-covalent interactions, the goregous ProLIF.

#### Concerns
1.  I am mostly a **functional programmer** , in that, I write scripts to answer specific biological questions using already established repos and for data visualization, and I am very much a beginner with OOP ( I have begun to learn to use MDAnalysis better, so thank you!)
2.  I don't know how to write unit tests or other core developer things yet :(

### Questions:

1. I have some functions to extend MDAnalysis for Antigen-Antibody analysis, including expanded definition for salt bridges, finding interactors across frames, and added functions in Seaborn to enhance the default documentation plots a bit. Would it be useful to the wider community if I submitted them, GSoC or not? If so, if there is post on how to submit code like this / pull request requirements, please let me know.
2. From the project ideas for this year, the On-the-fly-transformation project is exactly what I would be interested in. This would be my first time doing something like this, and I wanted to know if I will be a suitable candidate, given the concerns.

### Thank you for reading through the post!
Sampurna

**Comment by RMeli** [comment]
Thank you for participating in GSOC 2024 with us. Whether you were selected as a GSoC contributor or not, you are a valued member of the MDAnalysis community and we are excited to work with you outside of the GSoC program as well!
